src/handlers/permissions-grant.ts

```diff
@@ -29,22 +30,57 @@ export class PermissionsGrantHandler implements MethodHandler {
       return messageReplyFromError(e, 401);
     }
 
-    const { scope, conditions, ...propertiesToIndex } = message.descriptor;
-    const indexes: { [key: string]: string } = {
-      author: permissionsGrant.author!,
-      ...propertiesToIndex,
-    };
+    const indexes = PermissionsGrantHandler.constructMessageStoreIndexes(permissionsGrant);
 
     // If we have not seen this message before, store it
     const messageCid = await Message.getCid(message);
     const existingMessage = await this.messageStore.get(tenant, messageCid);
     if (existingMessage === undefined) {
       await this.messageStore.put(tenant, message, indexes);
-      await this.eventLog.append(tenant, messageCid);
+      const additionalIndexes = PermissionsGrantHandler.constructAdditionalEventLogIndexes(permissionsGrant);
+      await this.eventLog.append(tenant, messageCid, { ...indexes, ...additionalIndexes });
```

Comment:
Sanity: I am sure if you have thought this through already, but just in case: are you sure we don't need to worry about addition indexes overwriting the indexes? Is it either not possible or desirable?